src/libraries/Common/tests/TestUtilities/System/PlatformDetection.Unix.cs

```diff
@@ -204,11 +153,24 @@ private static Version ToVersion(string versionString)
             }
         }
 
-        private static DistroInfo GetDistroInfo() => new DistroInfo()
+        private static DistroInfo GetDistroInfo()
         {
-            Id = Microsoft.DotNet.PlatformAbstractions.RuntimeEnvironment.OperatingSystem,
-            VersionId = ToVersion(Microsoft.DotNet.PlatformAbstractions.RuntimeEnvironment.OperatingSystemVersion)
-        };
+            DistroInfo result = new DistroInfo();
+
+            foreach (string line in File.ReadAllLines("/etc/os-release"))
+            {
+                if (line.StartsWith("ID=", StringComparison.Ordinal))
+                {
+                    result.Id = line.Substring(3).Trim('"', '\'');
+                }
+                else if (line.StartsWith("VERSION_ID=", StringComparison.Ordinal))
```

Comment:
Should we throw or do something if the file doesn't contain the expected format?

Reply:
We never did before, we just blindly used whatever `PlatformAbstractions` returned. And if the file didn't contain the expected format, `PlatformAbstractions` would:

 return Id = `"Linux"`:

https://github.com/dotnet/runtime/blob/0ba41a844ae03bd1b0cfd06e00552be8e4b8f02a/src/installer/managed/Microsoft.DotNet.PlatformAbstractions/Native/PlatformApis.cs#L29

and Version = `string.Empty`:

https://github.com/dotnet/runtime/blob/0ba41a844ae03bd1b0cfd06e00552be8e4b8f02a/src/installer/managed/Microsoft.DotNet.PlatformAbstractions/Native/PlatformApis.cs#L46

I can default the Id to `"Linux"` and the version to what we get when we can't discover the version. Do you think that would make sense?

https://github.com/dotnet/runtime/blob/0ba41a844ae03bd1b0cfd06e00552be8e4b8f02a/src/libraries/Common/tests/TestUtilities/System/PlatformDetection.Unix.cs#L187-L189

Reply:
I see. Since that was product code, probably makes sense to follow what that was doing? Probably that is better than returning something no-sense and having non friendly test errors? So to me it kind of makes sense, I don't know how many tests rely on this though.

Reply:
I've updated this test code to do what the current product code is doing.